prov/efa/src/rdm/rxr_op_entry.c

```diff
@@ -1144,6 +1149,61 @@ int rxr_op_entry_prepare_to_post_read(struct rxr_op_entry *op_entry)
 	return 0;
 }
 
+/**
+ * @brief clone a packet from readcopy packet pool to ensure memory is registered.
+ *
+ * This function is applied on tx_entry->local_read_packet_entry.
+ *
+ * If the memory of the packet entry is not registered (this can happen when the
+ * packet is from OOO or unexp pakcet pool), this function will clone a packet
+ * from readcopy packet pool, release the original packet and update the tx_entry
+ * accordingly.
+ *
+ * Return value:
+ *
+ *     On success, return 0
+ *     On pack entry allocation failure, return -FI_EAGAIN
+ */
+static
+ssize_t rxr_tx_entry_prepare_local_read_pkt_entry_mr(struct rxr_op_entry *tx_entry)
```

Comment:
Do we used to have such clone to readcopy pool? Will it impact performance?

Reply:
It is a required step. so performance concern does not apply

Reply:
I see, so the original clone function is not removed because we have not got rid of read_entry yet, right?

Reply:
> I see, so the original clone function is not removed because we have not got rid of read_entry yet, right?

Yes. I plan to get rid of `rxr_read.c` and `rxr_read.h` via a separate commit, therefore no need to just delete 1 function